src/libraries/Microsoft.PowerFx.Core/Public/Values/InMemoryRecordValue.cs

```diff
@@ -24,6 +25,7 @@ public InMemoryRecordValue(IRContext irContext, IReadOnlyDictionary<string, Form
             Contract.Assert(IRContext.ResultType is RecordType);
 
             _fields = fields;
+            _sourceDict = fields as IDictionary<string, FormulaValue>;
```

Comment:
Why do we need to store the data twice?
And the names are misleading as the "source" is 'fields'

Reply:
What was the resolution for this comment? I have the same question.

Reply:
@bryancroteau-MSFT I kept _fields and changed it from IReadOnlyDictionary to IDictionary.